### PR TITLE
WIP: Registration page tweaks.

### LINKED ISSUE #48: More Javascript in the Registration page.
We need more javascript in the registration page for these functions:
- Check for username availability
- Check whether email is valid as per university
- Show "schools" belonging to the university only after the university has been selected.

### PR DESCRIPTION
**Pull request by digi0ps**
So far I have worked on these. These are mostly Javascript changes to the Registration page.

- [x]  The *School* select field updates dynamically based on the university the user selected.

- [x]  Now the username is checked asynchronously after the user has finished typing the username. 

- [x]  All the fields can we traversed by pressing enter.

- [ ] University specific email ( Though I think having users give their personal email will be better than their university email )

Fixes #48

### CODE REVIEW

File: RestApi/views.py [hunk 1]
```diff
@@ -2,21 +2,42 @@
...
 class TestPost(APIView):
     def post(self):
         pass
 
 
+class CheckUsername(APIView):
+    def post(self, request):
+        uname = request.data["username"]
+        try:
+            u = User.objects.get(username=uname)
```
Comment (jeremyphilemon) [inline RestApi/views.py]: @digi0ps Can you ignore cases when checking for the username? Jeremy is available while jeremy already exists. Our current registration model only accepts the creation of an unique username (case insensitive).

**Comment by jeremyphilemon** [comment]
@digi0ps Let's make university email mandatory during registration, since it's needed for verification.

## Or what about this?
- When the user clicks on register, we can ask him/her to enter his university email id. If it's valid and non-flagged, he can proceed to the registration form? (However, his uni's email will be attached to his user object)

File: RestApi/views.py (hunk 1 above)
Comment (digi0ps) [inline RestApi/views.py]: We are going all lowercase for username?

**Comment by digi0ps** [comment]
@jeremyphilemon Okay but I still think asking them there Uni email is not a good idea. 
I will list out the pros and cons.

Pros:
- The email be unique for each user. 
- We can blacklist their email and make sure that user doesn't return.

Cons:
- The user will feel the lack of privacy as they know that we have their university identity. This will suppress their opinion and won't give them complete anonymity. 
- We can't notify them or send digest emails as students rarely check their uni emails.
- What about the existing user database?

**Comment by jeremyphilemon** [comment]
> The user will feel the lack of privacy as they know that we have their university identity. This will suppress their opinion and won't give them complete anonymity.

We can mention that the email is used for verification purpose only and the said information will be protected no matter what happens. Regardless of the flagging system, this verification can prevent impersonation of other students too (think about the consequences of that too).

> We can't notify them or send digest emails as students rarely check their uni emails.

Exactly, the uni emails are for verification only while the other email id they provide during the actual registration can be used for sending them emails.

> What about the existing user database?

They're verified beta testers by default ¯\\_(ツ)_/¯

**Comment by digi0ps** [comment]
Okay what you are saying seems good to me! 
But won't it be irritating for users to enter two emails?

**Comment by jeremyphilemon** [comment]
I don't think we should compromise on the user's identity and safety for just another email field...

**Comment by digi0ps** [comment]
Okay! Where will we be asking the uni email?

File: RestApi/views.py (hunk 1 above)
Comment (jeremyphilemon) [inline RestApi/views.py]: Yeah?

**Comment by jeremyphilemon** [comment]
@digi0ps Like first we could show one field (centre of the screen) to enter their email id, then if it's valid the field moves to the left and the registration form enters from the right?

**Comment by jeremyphilemon** [comment]
@digi0ps Alright, this looks good. We can work on the university specific email by opening another branch!

**Pull request merged**